kmm/kmm-feed-data/src/commonMain/kotlin/com/gchristov/newsfeed/kmmfeeddata/RealFeedRepository.kt

```diff
@@ -103,4 +108,22 @@ internal class RealFeedRepository(
             }
         }
     }
-}
\ No newline at end of file
+
+    override suspend fun searchQuery(): String? =
+        withContext(dispatcher) {
+            if (sharedPreferences.contains(SEARCH_QUERY_PREFERENCES_KEY)) {
+                return@withContext sharedPreferences.getString(
```

Comment:
Does this return an optional? I imagine not since it has a default value. If not then we can simply the return here and remove the null else statement.